lib/sidekiq-status/server_middleware.rb

```diff
@@ -22,28 +22,34 @@ def initialize(opts = {})
     # @param [Array] msg job args, should have jid format
     # @param [String] queue queue name
     def call(worker, msg, queue)
-      # a way of overriding default expiration time,
-      # so worker wouldn't lose its data
-      # and it allows also to overwrite global expiration time on worker basis
-      if worker.respond_to? :expiration
-        if !worker.expiration && worker.respond_to?(:expiration=)
-          worker.expiration = @expiration
-        else
-          @expiration = worker.expiration
-        end
+
+      expiry = nil
+
+      # Determine the actual job class
+      klass = msg["args"][0]["job_class"] || msg["class"] rescue msg["class"]
+      job_class = klass.is_a?(Class) ? klass : Module.const_get(klass)
+
+      # Bypass unless this is a Sidekiq::Status::Worker job
+      unless job_class.ancestors.include?(Sidekiq::Status::Worker)
+        yield
+        return
       end
 
-      store_status worker.jid, :working,  @expiration
+      # Determine job expiration
+      expiry = job_class.new.expiration
```

Comment:
Instead `expiry = job_class.new.expiration`, We can use one item evaluation of `expiry` like `expiry = job_class.new.expiration || @expiration`

and Use it as `store_status worker.jid, :complete, @expiration`.